Source/WebCore/page/NavigatorLoginStatus.h

```diff
@@ -26,29 +26,34 @@
 #pragma once
 
 #include "Supplementable.h"
+#include "WebCore/RegistrableDomain.h"
+#include "dom/Document.h"
+#include <wtf/WeakPtr.h>
 
 namespace WebCore {
 
 class DeferredPromise;
 class Navigator;
-class NavigatorLoginStatus final : public Supplement<Navigator> {
+class Document;
+enum class IsLoggedIn : uint8_t;
+
+class NavigatorLoginStatus final : public Supplement<Navigator>, public CanMakeWeakPtr<NavigatorLoginStatus> {
     WTF_MAKE_FAST_ALLOCATED;
 public:
     explicit NavigatorLoginStatus(Navigator& navigator)
         : m_navigator(navigator)
     {
     }
-    static void setLoggedIn(Navigator&, Ref<DeferredPromise>&&);
-    static void setLoggedOut(Navigator&, Ref<DeferredPromise>&&);
+    static void setLoginStatus(Navigator&, const String& status, Ref<DeferredPromise>&&);
```

Comment:
Now that you're using an enum on IDL side, `status` could be a `LoginStatus` enum class here instead of a `String`.